format-common/src/main/java/io/cdap/plugin/format/connector/AbstractFileConnector.java

```diff
@@ -163,24 +214,77 @@ protected String getFullPath(String path) {
    * @param builder the builder of the spec
    */
   protected void setConnectorSpec(ConnectorSpecRequest request, ConnectorSpec.Builder builder) {
-    // no-op
+    //no op
+  }
+
+  /**
+   * Adds default sampling values to entity.
+   *
+   * @param entity entity to which add the sampling default values
+   * @param fileName name of the file. Used to guess correct format based on extension.
+   */
+  protected void addBrowseSampleDefaultValues(BrowseEntity.Builder entity, String fileName) {
+    entity.addProperty(SAMPLE_FORMAT_KEY, BrowseEntityPropertyValue.builder(
+      FileTypeDetector.detectFileFormat(FileTypeDetector.detectFileType(fileName)).name().toLowerCase(),
+      BrowseEntityPropertyValue.PropertyType.SAMPLE_DEFAULT).build());
+    entity.addProperty(SAMPLE_FILE_ENCODING_KEY, BrowseEntityPropertyValue.builder(
+      SAMPLE_FILE_ENCODING_DEFAULT, BrowseEntityPropertyValue.PropertyType.SAMPLE_DEFAULT).build());
   }
 
-  private ValidatingInputFormat getValidatingInputFormat(ConnectorContext context, String path) throws IOException {
+  /**
+   *
+   * @return Set of available sample properties
+   */
+  protected Set<BrowseEntityTypeInfo> getSampleProperties() {
+    return sampleProperties;
+  }
+
+  /**
+   * Receives a request and returns a map with only the valid sample properties.
+   *
+   * @param request Request containing the sample properties.
+   * @return a Map with chosen sample properties
+   */
+  protected Map<String, String> getSourceSampleProperties(ConnectorSpecRequest request) {
```

Comment:
So I believe this method is used to retrieve the spec for a plugin, then we shouldn't call it this name since the property returned will also be used by the sink. Just call this `getAdditionalSpecProperties`